classes/validation/ValidatorDateComparison.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/**
+ * @file classes/validation/ValidatorDateComparison.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class ValidatorDateComparison
+ *
+ * @ingroup validation
+ *
+ * @see Validator
+ *
+ * @brief Validation check for comparing with a given date
+ */
+
+namespace PKP\validation;
+
+use Carbon\Carbon;
+use DateTimeInterface;
+use Exception;
+use PKP\validation\ValidatorFactory;
+
+class ValidatorDateComparison extends Validator
+{
+    public const DATE_COMPARE_RULE_EQUAL = 'equal';
+    public const DATE_COMPARE_RULE_GREATER = 'greater';
+    public const DATE_COMPARE_RULE_LESSER = 'lesser';
+    public const DATE_COMPARE_RULE_GREATER_OR_EQUAL = 'greaterOrEqual';
+    public const DATE_COMPARE_RULE_LESSER_OR_EQUAL = 'lesserOrEqual';
+
+    protected DateTimeInterface|Carbon $comparingDate;
+
+    protected string $comparingRule;
+
+    protected array $validationRulesMapping = [
+        self::DATE_COMPARE_RULE_EQUAL               => 'date_equals',
+        self::DATE_COMPARE_RULE_GREATER             => 'after',
+        self::DATE_COMPARE_RULE_LESSER              => 'before',
+        self::DATE_COMPARE_RULE_GREATER_OR_EQUAL    => 'after_or_equal',
+        self::DATE_COMPARE_RULE_LESSER_OR_EQUAL     => 'before_or_equal',
+    ];
+
+    public function __construct(DateTimeInterface|Carbon $comparingDate, string $comparingRule)
+    {
+        if (!in_array($comparingRule, static::getComparingRules())) {
```

Comment:
What do you think about using an Enum for this, instead of runtime validation? We've applied this to a few other places in the code, and so far I like it.

Reply:
done refactoring to enum .